### DISCUSSION TITLE
Which version of org.apache.logging.log4j.log4j-slf4j2-impl supports org.slf4j.slf4j-api version 2.0.0 or more

### DISCUSSION THREAD

**Discussion by bharat-bhansali** [apache/logging-log4j2#2475] [Q&A] [answered]
We are mandated to upgrade org.slf4j.slf4j-api to version greater than 2.0.0 but when we upgrade it OSGI framework complaints that current version (2.17.1) of org.apache.logging.log4j.log4j-slf4j2-impl supports SLF4J version greater than equal to 1.7.0 and less than equal to 2.0.0

We checked all the versions of org.apache.logging.log4j.log4j-slf4j2-impl till Beta 3 but none of them supports SLF4J version greater than 2.0.0

Can you please help if their is a plan to release a supported version or any workaround?

**Comment by ppkarwasz** [comment, marked as answer]
@bbhansali1,

You are probably talking about the [`log4j-slf4j-impl`](https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j-impl) artifact (bundle symbolic name `org.apache.logging.log4j.slf4j-impl` before version 2.21.0 and `org.apache.logging.log4j.slf4j.impl` since then). That artifact will only support SLF4J 1.x.

SLF4J 2.x does not bring any major (i.e. breaking) changes to the public API, but it has breaking changes for providers, therefore we had to introduce a new artifact [`log4j-slf4j2-impl`](https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j2-impl) to support it. That is the artifact that you want to use.

BTW: I see that from a user perspective having two variants of the same SLF4J-to-Log4j API bridge is confusing. Would it help if we deprecate `log4j-slf4j-impl` and remove it from the distribution in the next two minor releases? Or do you still have some OSGi bundles that require SLF4J 1.x?

**Reply by bharat-bhansali** [reply]
Thanks @ppkarwasz for quick response. 

yes this was confusing as i downloaded all the versions of [log4j-slf4j-impl](https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j-impl) and none of them supported org.apache.logging.log4j.slf4j-impl version 2 or greater.

Either deprecating it or adding a note to point to [log4j-slf4j2-impl](https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j2-impl) will help.

Thanks,
Bharat

**Reply by ppkarwasz** [reply]
@bbhansali1,

Could you point us to the misleading documentation? We fixed the [/log4j/2.x/log4j-slf4j-impl](https://logging.apache.org/log4j/2.x/log4j-slf4j-impl) documentation page a long time ago, but there might be other references on our website that we missed.

**Reply by bharat-bhansali** [reply]
@ppkarwasz, 

what i meant is if you can add documentation on https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j-impl which points to https://mvnrepository.com/artifact/org.apache.logging.log4j/log4j-slf4j2-impl it will help users.
